aten/src/ATen/templates/Tensor.h

```diff
@@ -173,8 +173,8 @@ class CAFFE2_API Tensor {
     return impl_->strides();
   }
 #ifdef BUILD_NAMEDTENSOR
-  optional<DimnameList> names() const {
-    return impl::get_names(unsafeGetTensorImpl());
+  optional<DimnameList> opt_names() const {
+    return impl::get_opt_names(unsafeGetTensorImpl());
```

Comment:
Could it make sense to do something like returning a default value instead, e.g. a DimnameList containing all wildcards if empty?  It seems for device() and scalar_type() we raise an error or return UndefinedType respectively when not set.

Reply:
Yes, exactly, see https://github.com/pytorch/pytorch/pull/24914 . I've changed it so that:

1) `Tensor::opt_names` gives `Optional<DimnameList>` (this PR)
2) `Tensor::names` gives a default DimnameList containing all wildcards if empty. (https://github.com/pytorch/pytorch/pull/24914)

Reply:
Returning the default value is a great idea because it makes writing name inference rules a lot easier.